src/linear_algebra/affine_space.lean

```diff
@@ -227,41 +226,49 @@ by conv_lhs { rw [←vsub_vadd V1 p1 p2, map_vadd, vadd_vsub] }
 /-- Two affine maps are equal if they coerce to the same function. -/
 @[ext] lemma ext (f g : affine_map k V1 P1 V2 P2) (h : (f : P1 → P2) = g) : f = g :=
 begin
-  cases f,
-  cases g,
+  rcases f with ⟨f, f_linear, f_add⟩,
+  rcases g with ⟨g, g_linear, g_add⟩,
+  change f = g at h,
+  subst g,
   congr',
   ext v,
-  change f_to_fun = g_to_fun at h,
-  cases S1.nonempty with p,
-  have hvp : f_to_fun (v +ᵥ p) = g_to_fun (v +ᵥ p), { rw h },
-  rw [f_add, g_add, h] at hvp,
-  exact vadd_right_cancel V2 _ hvp
+  cases (add_torsor.nonempty V1 : nonempty P1) with p,
+  apply vadd_right_cancel V2 (f p),
+  erw [← f_add, ← g_add]
 end
 
+variables (k V1 P1)
+
 /-- Identity map as an affine map. -/
 def id : affine_map k V1 P1 V1 P1 :=
 { to_fun := id,
   linear := linear_map.id,
-  add := λ p v, rfl }
+  map_vadd' := λ p v, rfl }
 
 /-- The identity affine map acts as the identity. -/
 @[simp] lemma id_apply (p : P1) : (id k V1 P1) p = p := rfl
```

Comment:
In principle, `P1` could be implicit here, right?

Reply:
Done.